database.go

```diff
@@ -221,6 +221,34 @@ func (m *Measurement) addSeries(s *Series) bool {
 	return true
 }
 
+// removeSeries will remove a series from the measurementIndex. Returns true if already removed
+func (m *Measurement) removeSeries(seriesID uint32) bool {
+	if _, ok := m.seriesByID[seriesID]; !ok {
+		return true
+	}
+	s := m.seriesByID[seriesID]
+	tagset := string(marshalTags(s.Tags))
+
+	delete(m.series, tagset)
+	delete(m.seriesByID, seriesID)
+
+	var ids []uint32
+	for _, id := range m.seriesIDs {
+		if id != seriesID {
+			ids = append(ids, id)
+		}
+	}
+	m.seriesIDs = ids
+	sort.Sort(m.seriesIDs)
+
+	// add this series id to the tag index on the measurement
+	for k, _ := range s.Tags {
+		delete(m.seriesByTagKeyValue, k)
```

Comment:
This is actually going to break things. If you look at that map it's `tagKey -> tagValue -> seriesIDs`. What you want to do is to look up each tag key/value pair in that map and remove the series id from that collection. If after removal, the collection is empty, then you can delete the tagValue. If after removal of the tagValue the parent collection is empty, then you can remove the tagKey